scale/job/test/test_views.py

```diff
@@ -921,4193 +596,1561 @@ def test_superseded(self):
-        self.assertEqual(result['results'][0]['is_operational'], self.job_type2.is_operational)
 
-    def test_sorting(self):
-        """Tests custom sorting."""
+    def test_is_active(self):
+        """Tests successfully calling the job types view filtered by inactive state."""
 
-        url = '/%s/job-types/?order=priority' % self.api
-        response = self.client.get(url)
+        url = '/%s/job-type-names/?is_active=false' % self.api
+        response = self.client.generic('GET', url)
         self.assertEqual(response.status_code, status.HTTP_200_OK, response.content)
-
         result = json.loads(response.content)
+
         self.assertEqual(len(result['results']), 2)
-        self.assertEqual(result['results'][0]['id'], self.job_type2.id)
-        self.assertEqual(result['results'][0]['name'], self.job_type2.name)
-        self.assertEqual(result['results'][0]['version'], self.job_type2.version)
 
-    def test_reverse_sorting(self):
-        """Tests custom sorting in reverse."""
+    def test_is_system(self):
+        """Tests successfully calling the job types view filtered by system status."""
 
-        url = '/%s/job-types/?order=-mem_const_required' % self.api
-        response = self.client.get(url)
+        url = '/%s/job-type-names/?is_system=false' % self.api
+        response = self.client.generic('GET', url)
         self.assertEqual(response.status_code, status.HTTP_200_OK, response.content)
 
         result = json.loads(response.content)
-        self.assertEqual(len(result['results']), 2)
-        self.assertEqual(result['results'][0]['id'], self.job_type2.id)
-        self.assertEqual(result['results'][0]['name'], self.job_type2.name)
-        self.assertEqual(result['results'][0]['version'], self.job_type2.version)
+        self.assertEqual(len(result['results']), 3)
 
-    def test_create(self):
-        """Tests creating a new job type."""
-        url = '/%s/job-types/' % self.api
-        json_data = {
-            'name': 'job-type-post-test',
-            'version': '1.0.0',
-            'title': 'Job Type Post Test',
-            'description': 'This is a test.',
-            'priority': '1',
-            'interface': {
-                'version': '1.0',
-                'command': 'test_cmd',
-                'command_arguments': 'test_arg',
-                'input_data': [],
-                'output_data': [],
-                'shared_resources': [],
+        url = '/%s/job-type-names/?is_system=true' % self.api
+        response = self.client.generic('GET', url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.content)
+
+        result = json.loads(response.content)
+        self.assertEqual(len(result['results']), 1)
+
+class TestJobTypesPostViewV6(TestCase):
+
+    api = 'v6'
+
+    def setUp(self):
+        django.setup()
+
+        self.manifest = job_test_utils.COMPLETE_MANIFEST
+
+        self.interface = {
```

Comment:
Is there a test that uses the old style interface?

Reply:
No. They have all been converted to v6 interfaces. Removed the unused definitions.